tsdb/head.go

```diff
@@ -2040,106 +2040,118 @@ func (h *Head) updateWALReplayStatusRead(current int) {
 
 func GenerateTestHistograms(n int) (r []*histogram.Histogram) {
 	for i := 0; i < n; i++ {
-		h := histogram.Histogram{
-			Count:         10 + uint64(i*8),
-			ZeroCount:     2 + uint64(i),
-			ZeroThreshold: 0.001,
-			Sum:           18.4 * float64(i+1),
-			Schema:        1,
-			PositiveSpans: []histogram.Span{
-				{Offset: 0, Length: 2},
-				{Offset: 1, Length: 2},
-			},
-			PositiveBuckets: []int64{int64(i + 1), 1, -1, 0},
-			NegativeSpans: []histogram.Span{
-				{Offset: 0, Length: 2},
-				{Offset: 1, Length: 2},
-			},
-			NegativeBuckets: []int64{int64(i + 1), 1, -1, 0},
-		}
-		if i > 0 {
-			h.CounterResetHint = histogram.NotCounterReset
-		}
-		r = append(r, &h)
+		r = append(r, GenerateTestHistogram(i))
 	}
 	return r
 }
 
+func GenerateTestHistogram(i int) *histogram.Histogram {
+	h := histogram.Histogram{
+		Count:         10 + uint64(i*8),
+		ZeroCount:     2 + uint64(i),
+		ZeroThreshold: 0.001,
+		Sum:           18.4 * float64(i+1),
+		Schema:        1,
+		PositiveSpans: []histogram.Span{
+			{Offset: 0, Length: 2},
+			{Offset: 1, Length: 2},
+		},
+		PositiveBuckets: []int64{int64(i + 1), 1, -1, 0},
+		NegativeSpans: []histogram.Span{
+			{Offset: 0, Length: 2},
+			{Offset: 1, Length: 2},
+		},
+		NegativeBuckets: []int64{int64(i + 1), 1, -1, 0},
+	}
+	if i > 0 {
+		h.CounterResetHint = histogram.NotCounterReset
+	}
```

Comment:
Looking from a perspective of exported functions, this should go in `GenerateTestHistograms`. 

`GenerateTestHistogram`, from an external perspective, just gives a test histogram, and the user has to decide the counter reset header.

Reply:
makes sense, updated